src/java.base/share/classes/java/lang/Comparable.java

```diff
@@ -64,8 +64,8 @@
  *
  * Virtually all Java core classes that implement {@code Comparable} have natural
  * orderings that are consistent with equals.  One exception is
- * {@code java.math.BigDecimal}, whose natural ordering equates
- * {@code BigDecimal} objects with equal values and different precisions
+ * {@link java.math.BigDecimal}, whose {@linkplain java.math.BigDecimal#compareTo natural ordering} equates
+ * {@code BigDecimal} objects with equal numerical values and different representations
```

Comment:
Nothing here talks about the representation of BigDecimal. I think it would be preferable to say that in BigDecimal, the `equals()` method considers 4.0 and 4.00 unequal because they have different precisions; however, `compareTo()` considers them equal because it considers only their numerical values.

Reply:
In BigDecimal, I'll add a sentence:

" The results of methods like {@link scale} and {@link unscaledValue} will differ for numerically equal values with different representations."

Talking about "precision" and BigDecimal is more appropriate for pre-JSR 13 BigDecimal since values like zero can have the same precision (1 digit) but many different scales.